### PR TITLE
Extending unique to unique_count and value_count functions to numerical.py and series.py

### PR DESCRIPTION
**Pull request by dantegd**
Extending unique() functionality to unique_count() and value_count() in pygdf code, needed for some current usage scenarios.

Functionality similar to pandas or clickhouse functions, in pygdf.

Will port whole functionality to libgdf later.

### CODE REVIEW

File: pygdf/series.py [hunk 1]
```diff
@@ -7,6 +9,7 @@
 import numpy as np
 
 from . import cudautils, formatting
+from . import dataframe
```
Comment (kkraus14) [inline pygdf/series.py]: This is a circular import as dataframe imports series, but I believe you aren't using this so it can be removed.

**Comment by kkraus14** [comment]
This is a sort based approach which would be less efficient than a hash based approach (unless data is already sorted or needs to be sorted anyway), so we should plan to have a parameter to specify whether to use the sort based method or hash based method in the future.

**Comment by kkraus14** [comment]
Needs test for `.unique_count()`

File: pygdf/series.py (hunk 1 above)
Comment (dantegd) [reply pygdf/series.py]: The import was a remnant of prior code I was trying, forgot about it, will remove it.

Agree with needing the parameter for sort (similar to pandas actually), will add it.

Test for unique_count is already in test_stats (line 59), the same function for unique and value counts:

```assert len(set(arr[mask])) == sr.unique_count()```

Didn't want to add new functions for testing three things that share most of the code, let me know if its alright.

File: pygdf/numerical.py [hunk 2]
```diff
@@ -141,17 +141,18 @@ def _unique_segments(self):
         segs, begins = cudautils.find_segments(sortedvals)
         return segs, sortedvals
 
-    def unique(self):
+    def unique(self, type='hash'):
```
Comment (kkraus14) [inline pygdf/numerical.py]: We should add the templating here for checking the type and if someone passes "hash" it raises a not implemented exception, and the default should be "sort" for now.

File: pygdf/numerical.py [hunk 3]
```diff
@@ -141,17 +141,18 @@ def _unique_segments(self):
...
+        # type variable will indicate what algorithm to use to calculate unique, not used right now
         segs, sortedvals = self._unique_segments()
         # gather result
         out = cudautils.gather(data=sortedvals, index=segs)
         return self.replace(data=Buffer(out), mask=None)
 
-    def unique_count(self):
+    def unique_count(self, type='hash'):
```
Comment (kkraus14) [inline pygdf/numerical.py]: Same as above

File: pygdf/numerical.py [hunk 4]
```diff
@@ -141,17 +141,18 @@ def _unique_segments(self):
...
         segs, _ = self._unique_segments()
         return len(segs)
 
-    def value_count(self):
+    def value_count(self, type='hash'):
```
Comment (kkraus14) [inline pygdf/numerical.py]: Same as above

Comment (kkraus14) [reply pygdf/numerical.py]: This should also be `value_counts` instead of `value_count`

File: pygdf/series.py [hunk 5]
```diff
@@ -700,14 +702,43 @@ def unique_k(self, k):
...
+        if type is not 'sort':
+            msg = 'non sort based unique_count() not implemented yet'
+            raise NotImplementedError(msg)
+        if self.null_count == len(self):
+            return 0
+        return self._column.unique_count(type=type)
+        # return len(self._column.unique())
+
+
+    def value_count(self, type='sort'):
```
Comment (kkraus14) [inline pygdf/series.py]: This should be `value_counts` instead of `value_count`

**Comment by kkraus14** [comment]
Build is now succeeding, looks good to me, merging.

**Pull request merged**